tests/test_models/test_holt_winters_model.py

```diff
@@ -119,3 +123,155 @@ def test_get_model_after_training(example_tsds, etna_model_class, expected_class
     assert_model_equals_loaded_original(model=model, ts=example_tsds, transforms=[], horizon=3)
+
+
+@pytest.fixture()
+def multi_trend_dfs(multitrend_df):
+    df = multitrend_df.copy()
+    df.columns = df.columns.droplevel("segment")
+    df.reset_index(inplace=True)
+    df["target"] += 10 - df["target"].min()
+
+    return df.iloc[:-9], df.iloc[-9:]
+
+
+@pytest.fixture()
+def seasonal_dfs():
+    target = pd.Series(
+        [
+            41.727458,
+            24.041850,
+            32.328103,
+            37.328708,
+            46.213153,
+            29.346326,
+            36.482910,
+            42.977719,
+            48.901525,
+            31.180221,
+            37.717881,
+            40.420211,
+            51.206863,
+            31.887228,
+            40.978263,
+            43.772491,
+            55.558567,
+            33.850915,
+            42.076383,
+            45.642292,
+            59.766780,
+            35.191877,
+            44.319737,
+            47.913736,
+        ],
+        index=pd.period_range(start="2005Q1", end="2010Q4", freq="Q"),
+    )
+
+    df = pd.DataFrame(
+        {
+            "timestamp": target.index.to_timestamp(),
+            "target": target.values,
+        }
+    )
+
+    return df.iloc[:-9], df.iloc[-9:]
+
+
+def test_check_mul_components_not_fitted_error():
+    model = _HoltWintersAdapter()
+    with pytest.raises(ValueError, match="This model is not fitted!"):
+        model._check_mul_components()
+
+
+def test_predict_components_not_fitted_error(seasonal_dfs):
+    _, df = seasonal_dfs
+    model = _HoltWintersAdapter()
+    with pytest.raises(ValueError, match="This model is not fitted!"):
+        model.predict_components(df=df)
+
+
+def test_forecast_components_not_fitted_error(seasonal_dfs):
+    _, df = seasonal_dfs
+    model = _HoltWintersAdapter()
+    with pytest.raises(ValueError, match="This model is not fitted!"):
+        model.forecast_components(df=df)
+
+
+@pytest.mark.parametrize("trend,trend_error", (("mul", True), ("add", False), (None, False)))
+@pytest.mark.parametrize("seasonal,seasonal_error", (("mul", True), ("add", False), (None, False)))
+def test_check_mul_components(seasonal_dfs, trend, trend_error, seasonal, seasonal_error):
+    _, df = seasonal_dfs
+    model = _HoltWintersAdapter(trend=trend, seasonal=seasonal)
+    model.fit(df, [])
+
+    if trend_error or seasonal_error:
+        context = pytest.raises(ValueError, match="Forecast decomposition is only supported for additive components!")
+    else:
+        context = nullcontext()
+
+    with context:
+        model._check_mul_components()
+
+
+@pytest.mark.parametrize("trend,trend_component", (("add", ["target_component_trend"]), (None, [])))
+@pytest.mark.parametrize("seasonal,seasonal_component", (("add", ["target_component_seasonality"]), (None, [])))
+def test_predict_components_names(seasonal_dfs, trend, trend_component, seasonal, seasonal_component):
```

Comment:
Let's make one test of each pair(names + sum up to target) for predict/forecast, parametrize the prediction method like [here](https://github.com/tinkoff-ai/etna/blob/93f89590c2ca549431c8aa2d2094907f2f81377a/tests/test_models/test_mixins.py#L57)